core/src/main/java/org/apache/brooklyn/core/mgmt/internal/AbstractManagementContext.java

```diff
@@ -372,6 +377,27 @@ public BrooklynProperties getBrooklynProperties() {
         return configMap;
     }
 
+    public StringConfigMap getConfigWithExternalConfigResolved() {
+        final ExternalConfigSupplierRegistry externalConfigProviderRegistry = getExternalConfigProviderRegistry();
+        if (externalConfigProviderRegistry == null)
+            return configMap; // no external config available yet, so return config without resolving
```

Comment:
May be set external config values to null to prevent using the raw string or even throw as the caller would expect them to be resolved already?